DSI/globus_gridftp_server_iRODS.cpp

```diff
@@ -2075,6 +2076,8 @@ globus_l_gfs_iRODS_recv(
 
         if(transfer_info->pathname == nullptr)
         {
+            free(collection);
```

Comment:
Wasn't sure if we were leaning more C or C++ in this file, but prefer `std::free` if the latter.

Reply:
> Wasn't sure if we were leaning more C or C++ in this file, but prefer `std::free` if the latter.

I went ahead and made this change for all free() calls in that one method.  I figured keeping it consistent at least within that method is preferrable.